### PR TITLE
P5 Travel App

### PR DESCRIPTION
**Pull request by leemoispace**
P5 Capstone-Travel-App using several APIs together

[project rubics](https://learn-udacity.top/fewd586261/Front%20End%20Web%20Developer%20v5.0.0/Part%2007-Module%2001-Lesson%2001_Project%20FEND%20Capstone%20-%20Travel%20App/Project%20Rubric%20-%20FEND%20Capstone%20-%20Travel%20App.html)

https://user-images.githubusercontent.com/2882655/182943168-fbc40b2b-ddbc-4b3c-b5a1-417345fa043a.mp4

### CODE REVIEW

File: p5-travel-app/src/client/index.js [hunk 1]
```diff
@@ -5,7 +5,11 @@
 //import functions
 
 //import style
-import "./styles/index.scss";
+//import "bootstrap/dist/css/bootstrap.min.css";
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/index.js]: if you want to commit unfinished code, please add “// TODO” or “// FIX ME” on the top of it

**Review by Martian2Lee — CHANGES_REQUESTED** [review]

File: p5-travel-app/src/client/index.js [hunk 2]
```diff
@@ -3,14 +3,40 @@
 // styles referenced in html/css
 
 //import functions
+import {getGeoCity} from './js/geoNamesApi';
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/index.js]: not prettiered

File: p5-travel-app/src/client/js/weatherApi.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const weatherKey = "b3432a956d7b45a18ba374d67c837a2e";
+
+export async function getWeatherForecast(latitude, longitude) {
+  const endpoint =
+    "https://api.weatherbit.io/v2.0/forecast/daily?" +
+    `lat=${latitude}&lon=${longitude}&key=` +
+    weatherKey;
+  //console.log(endpoint);
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/js/weatherApi.js]: remove console

File: p5-travel-app/src/client/js/weatherApi.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ endpoint: endpoint }),
+    });
+    if (res.ok) {
+      const data = await res.json();
+      return data;
+    }
+  } catch (error) {
+    console.log(error);
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/js/weatherApi.js]: console warm

File: p5-travel-app/src/client/js/weatherApi.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+const weatherKey = "b3432a956d7b45a18ba374d67c837a2e";
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/js/weatherApi.js]: not in code, save in env

File: p5-travel-app/src/client/js/geoNamesApi.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    const res = await fetch(dataUrl);
+    if (res.ok) {
+      const city = {};
+      const data = await res.json();
+      //console.log(data);
+      city.latitude = data.geonames[0].lat;
+      city.longitude = data.geonames[0].lng;
+      city.population = data.geonames[0].population;
+      city.countryCode = data.geonames[0].countryCode;
+      console.log(city);
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/js/geoNamesApi.js]: keep console in your local, don’t commit this line

File: p5-travel-app/src/client/js/geoNamesApi.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    city +
+    "&username=" +
+    username +
+    "&style=full";
+  try {
+    const res = await fetch(dataUrl);
+    if (res.ok) {
+      const city = {};
+      const data = await res.json();
+      //console.log(data);
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/js/geoNamesApi.js]: Remove

File: p5-travel-app/src/client/js/form.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  const date = document.getElementById("date_end").value.split("-");
+  return date.join("-");
+};
+
+const countdown = (start, end) => {
+  const tripStart = Date.parse(start);
+  const tripEnd = Date.parse(end);
+  const countdown = tripEnd - tripStart;
+  const daysLeft = Math.ceil(countdown / 86400000);
+  //console.log(daysLeft);
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/js/form.js]: Remove

File: p5-travel-app/src/client/js/form.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+const getCity = () => {
+  const city = document.getElementById("city").value;
+  city = city.toLowerCase();
+  city = city[0].toUpperCase() + city.slice(1);
+  console.log(city);
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/js/form.js]: Remove

**Review by Martian2Lee — CHANGES_REQUESTED** [review]

File: p5-travel-app/src/client/js/pixabayApi.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+export async function getImageURL(city, country) {
+  const queryCity = `&q=${city}&image_type=photo&pretty=true&category=places`;
+  const queryCountry = `&q=${country}&image_type=photo&pretty=true&category=places`;
+  const cityEndpoint = "https://pixabay.com/api/?key=" + pixabayKey + queryCity;
+  const countryEndpoint =
+    "https://pixabay.com/api/?key=" + pixabayKey + queryCountry;
+  try {
+    let response = await fetch(cityEndpoint);
+    //console.log(response);
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/js/pixabayApi.js]: remove

**Review by Martian2Lee — CHANGES_REQUESTED** [review]

File: p5-travel-app/src/server/server.js [hunk 11]
```diff
@@ -13,3 +26,35 @@ app.use(bodyParser.json());
...
+  if (req.body.endpoint !== " ") {
+    const endpoint = req.body.endpoint;
+    try {
+      const response = await fetch(endpoint);
+      if (response.ok) {
+        const jsonRes = await response.json();
+        res.status(201).send(jsonRes);
+      }
+    } catch (error) {
+      console.log(error);
```
Comment (Martian2Lee) [inline p5-travel-app/src/server/server.js]: console warn

**Review by Martian2Lee — CHANGES_REQUESTED** [review]

File: p5-travel-app/src/client/js/model.js [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+  document.querySelector(".caption").style.top = "5vh";
+  $("#tripModal").modal("toggle");
+
+  const tripStart = getTripDate(trip.start);
+  const tripEnd = getTripDate(trip.end);
+  const daysLeft = countdown(new Date(), trip.start);
+  const weather = getWeatherInfo(trip.weatherForecast, daysLeft, tripStart);
+  const div = document.createElement("div");
+  div.classList.add("col-md-4");
+  //section.appendChild(div);
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/js/model.js]: remove

File: p5-travel-app/src/client/js/model.js [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+  const tripEnd = getTripDate(trip.end);
+  document.querySelectorAll(".media_heading")[1].innerText = tripStart;
+  document.querySelectorAll(".media_heading")[2].innerText = tripEnd;
+  document.querySelectorAll(".media_heading")[3].innerText = `${countdown(
+    trip.start,
+    trip.end
+  )} days`;
+
+  // Display trip images
+  // const imageURL = await getTripImageURL(images);
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/js/model.js]: remove

**Review by Martian2Lee — CHANGES_REQUESTED** [review]

File: p5-travel-app/src/client/index.js [hunk 14]
```diff
@@ -6,9 +6,12 @@
 import { getGeoCity } from "./js/geoNamesApi";
 import { getWeatherForecast } from "./js/weatherApi";
 import { getCity, getTripStart, getTripEnd } from "./js/form";
+import { showModal, recentTrip } from "./js/model";
+import { getCountryInfo } from "./js/restCountriesApi";
 
 //import style
 import "./styles/style.scss";
+// import "bootstrap/dist/css/bootstrap.min.css";
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/index.js]: remove

**Review by Martian2Lee — CHANGES_REQUESTED** [review]

File: p5-travel-app/src/client/js/restCountriesApi.js [hunk 15]
```diff
@@ -1,12 +1,14 @@
+//https://restcountries.com/
 export async function getCountryInfo(countryCode) {
-  const endpoint = `https://restcountries.eu/rest/v2/alpha/${countryCode}`;
+  const endpoint = `https://restcountries.com/v3.1/alpha/${countryCode}`;
   try {
     const res = await fetch(endpoint);
     if (res.ok) {
       const data = await res.json();
+      console.log(data);
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/js/restCountriesApi.js]: remove

File: p5-travel-app/src/client/index.js [hunk 16]
```diff
@@ -35,6 +36,7 @@ const handleSearch = async (e) => {
   trip.country = countryInfo.name;
   trip.countryFlag = countryInfo.flag;
   trip.image = await getImageURL(trip.city, trip.country);
+  console.log(trip);
```
Comment (Martian2Lee) [inline p5-travel-app/src/client/index.js]: remove

File: p5-travel-app/webpack.dev.js [hunk 17]
```diff
@@ -33,12 +33,13 @@ module.exports = {
     rules: [
       {
         test: "/.js$/",
-        exclude: /node_modules/,
+        // exclude: /node_modules/,
```
Comment (Martian2Lee) [inline p5-travel-app/webpack.dev.js]: remove or add comment

**Review by Martian2Lee — CHANGES_REQUESTED** [review]

File: p5-travel-app/webpack.dev.js (hunk 17 above)
Comment (leemoispace) [reply p5-travel-app/webpack.dev.js]: done, clean it in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/client/index.js (hunk 16 above)
Comment (leemoispace) [reply p5-travel-app/src/client/index.js]: done, clean it in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/client/js/restCountriesApi.js (hunk 15 above)
Comment (leemoispace) [reply p5-travel-app/src/client/js/restCountriesApi.js]: done, clean it in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/client/index.js (hunk 14 above)
Comment (leemoispace) [reply p5-travel-app/src/client/index.js]: done, clean it in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/client/js/model.js (hunk 13 above)
Comment (leemoispace) [reply p5-travel-app/src/client/js/model.js]: done, clean it in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/client/js/model.js (hunk 12 above)
Comment (leemoispace) [reply p5-travel-app/src/client/js/model.js]: done, clean it in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/server/server.js (hunk 11 above)
Comment (leemoispace) [reply p5-travel-app/src/server/server.js]: done in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/client/js/pixabayApi.js (hunk 10 above)
Comment (leemoispace) [reply p5-travel-app/src/client/js/pixabayApi.js]: done in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/client/js/form.js (hunk 9 above)
Comment (leemoispace) [reply p5-travel-app/src/client/js/form.js]: done in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/client/js/form.js (hunk 8 above)
Comment (leemoispace) [reply p5-travel-app/src/client/js/form.js]: done in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/client/js/geoNamesApi.js (hunk 7 above)
Comment (leemoispace) [reply p5-travel-app/src/client/js/geoNamesApi.js]: done in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/client/js/geoNamesApi.js (hunk 6 above)
Comment (leemoispace) [reply p5-travel-app/src/client/js/geoNamesApi.js]: done in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/client/js/weatherApi.js (hunk 5 above)
Comment (leemoispace) [reply p5-travel-app/src/client/js/weatherApi.js]: done in [4b6e967](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/4b6e96742131ae1774f8acaffc88a20f0b7bf430)

File: p5-travel-app/src/client/js/weatherApi.js (hunk 4 above)
Comment (leemoispace) [reply p5-travel-app/src/client/js/weatherApi.js]: done in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/client/js/weatherApi.js (hunk 3 above)
Comment (leemoispace) [reply p5-travel-app/src/client/js/weatherApi.js]: done in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/client/index.js (hunk 2 above)
Comment (leemoispace) [reply p5-travel-app/src/client/index.js]: done in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

File: p5-travel-app/src/client/index.js (hunk 1 above)
Comment (leemoispace) [reply p5-travel-app/src/client/index.js]: done in [77541ac](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/9/commits/77541acca892e61c7f81d34c2e31e59fbb1a7011)

**Review by Martian2Lee — COMMENTED** [review]
Brillian README!

**Review by Martian2Lee — APPROVED** [review]
🚀

**Pull request merged**